velox/exec/HashTable.cpp

```diff
@@ -524,21 +524,19 @@ void HashTable<ignoreNullKeys>::initializeNewGroups(HashLookup& lookup) {
 template <bool ignoreNullKeys>
 void HashTable<ignoreNullKeys>::allocateTables(uint64_t size) {
   VELOX_CHECK(bits::isPowerOfTwo(size), "Size is not a power of two: {}", size);
-  if (tags_) {
-    free(tags_);
-    tags_ = nullptr;
-  }
-  if (table_) {
-    free(table_);
-    table_ = nullptr;
-  }
   if (size > 0) {
     size_ = size;
     sizeMask_ = size_ - 1;
     sizeBits_ = __builtin_popcountll(sizeMask_);
-    tags_ = reinterpret_cast<uint8_t*>(aligned_alloc(64, size_));
+    constexpr auto kPageSize = memory::MappedMemory::kPageSize;
+    auto numPages = bits::roundUp(size * 9, kPageSize) / kPageSize;
```

Comment:
where does this 9 come from? Is this to account for the tags at the end of the new allocation? Could use a small comment to describe what's going on .